### PR TITLE
🐛 fix rpc tests and docs

### PR DESCRIPTION
**Pull request by EvolveArt**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

X Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
X Documentation content changes
X Testing
- Other (please describe):

## What is the current behavior?

Resolves: #1119 

- Not enough tests
- Ignored tests for Cairo 1 contracts declaration 
- Outdated `rpc-contribution.md` docs

## What is the new behavior?

- Updates contribution doc
- Add more tests based on current issues and js scripts
- Remove `#[ignore]` on ignored tests

## Does this introduce a breaking change?

No

### CODE REVIEW

File: starknet-rpc-test/src/lib.rs [hunk 1]
```diff
@@ -140,29 +140,46 @@ fn get_free_port() -> Result<u16, TestError> {
     Err(TestError::NoFreePorts)
 }
 
+fn get_free_p2p_port() -> Result<u16, TestError> {
```
Comment (d-roak) [inline starknet-rpc-test/src/lib.rs:143]: why do we need to set the p2p port? it's setted as a random one if the default is already occupied and we are not querying it

File: starknet-rpc-test/src/lib.rs [hunk 2]
```diff
@@ -140,29 +140,46 @@ fn get_free_port() -> Result<u16, TestError> {
...
         let child_handle = Command::new("cargo")
 		// Silence Madara stdout and stderr
 		.stdout(Stdio::null())
-		.stderr(Stdio::null())
+		.stderr(if madara_log == "true" { Stdio::inherit() } else { Stdio::null() })
 		.args([
 			"run",
 			"--release",
 			"--",
 			"--alice",
```
Comment (d-roak) [inline starknet-rpc-test/src/lib.rs:175]: we can remove this, is already setted with `--dev`

File: starknet-rpc-test/src/lib.rs [hunk 3]
```diff
@@ -140,29 +140,46 @@ fn get_free_port() -> Result<u16, TestError> {
...
 			"--sealing=manual",
 			&format!("--execution={execution}"),
-			"--chain=dev",
+			"--dev",
 			"--tmp",
-			&format!("--rpc-port={free_port}")
+			&format!("--port={free_p2p_port}"),
+			&format!("--rpc-port={free_port}"),
+            &format!("--madara-path=/tmp/{}", free_p2p_port)
```
Comment (d-roak) [inline starknet-rpc-test/src/lib.rs:182]: if we use this, we should remove the `--tmp` flag

File: .github/workflows/linters.yml [hunk 4]
```diff
@@ -6,14 +6,6 @@ on:
   workflow_call:
 
 jobs:
-  prettier:
```
Comment (d-roak) [inline .github/workflows/linters.yml:9]: why are we removing this?

File: starknet-rpc-test/src/lib.rs [hunk 5]
```diff
@@ -129,45 +134,60 @@ impl Drop for MadaraClient {
...
 			"--",
-			"--alice",
 			"--sealing=manual",
 			&format!("--execution={execution}"),
-			"--chain=dev",
-			"--tmp",
-			&format!("--rpc-port={free_port}")
+			"--dev",
+            "--tmp",
+			&format!("--port={p2p_port}"),
```
Comment (d-roak) [inline starknet-rpc-test/src/lib.rs:184]: why are we setting the p2p port? do we need reproducibility on those?

Comment (EvolveArt) [reply starknet-rpc-test/src/lib.rs:184]: Yes and also in case the default is not available

**Review by d-roak — APPROVED** [review]

**Pull request merged**